src/room/GroupCallView.tsx

```diff
@@ -134,6 +134,13 @@ export const GroupCallView: FC<Props> = ({
   const latestMuteStates = useRef<MuteStates>();
   latestMuteStates.current = muteStates;
 
+  useEffect(() => {
+    // skip lobby if we are not in widget mode;
+    if (!widget && skipLobby) {
+      void enterRTCSession(rtcSession, perParticipantE2EE);
+    }
+  }, [perParticipantE2EE, rtcSession, skipLobby]);
```

Comment:
In the interest of keeping widget mode and non-widget mode on similar code paths where possible… could this branch and the `if (widget && !preload && skipLobby)` branch be combined into one? It seems like modifying the effect just below this to look like:

```typescript
if (skipLobby) {
  if (widget && preload) {
    // In preload mode without lobby we wait for a join action before entering
    ...
  } else {
    // No lobby and no preload: we enter the rtc session right away
    ... enterRTCSession(rtcSession, perParticipantE2EE) ...
  }
}
```

could work?